src/pudl/transform/eia861.py

```diff
@@ -1020,6 +1022,55 @@ def _harvest_associations(dfs: list[pd.DataFrame], cols: list[str]) -> pd.DataFr
     return assn
 
 
+def _combine_88888_values(df: pd.DataFrame, idx_cols: list[str]) -> pd.DataFrame:
+    """Combine 88888 utility_id_values.
+
+    This function sums rows with a utility_id_eia of 88888 into a single row by
+    primary key. It drops rows with a utility_id_eia of 88888 if there are non-numeric
+    columns with different values that are impossible to combine. E.g.: boolean
+    columns where one value is Y and the other is N.
+
+    Args:
+        df: The dataframe to process.
+        idx_cols: The index (primary key) columns of the input dataframe. They must
+            identify unique records in the input data.
+
+    Returns:
+        A dataframe with the combined or dropped 88888 utiltiy_id_eia values.
+    """
+    if 88888 in df.utility_id_eia.unique():
+        non_num_cols = (
+            df.set_index(idx_cols).select_dtypes(exclude="number").columns.tolist()
+        )
+
+        def custom_group_agg(group):
+            if len(group) <= 1:
+                return group
+            # If there are no non-numeric columns, or all non-numeric columns have the same value,
+            # Sum the numeric columns and return the result.
+            if (
+                len(non_num_cols) == 0
+                or ((group[non_num_cols] == group[non_num_cols].iloc[0]).all()).all()
+            ):
+                group = group.set_index(idx_cols)
+                non_num_group = group[non_num_cols].iloc[[0]]
+                num_group = group.drop(columns=non_num_cols).groupby(idx_cols).sum()
+                return pd.concat([non_num_group, num_group], axis=1).reset_index()
+            # Exclude rows with 88888 utility_id_eia that can't be combined due to different values in
+            # non-numeric columns.
+            return None
```

Comment:
is there a rough idea of how many rows we have to exclude? /would that be important for users to know?

Reply:
That's a good point. Might want to add a check that there aren't more than a certain number getting dropped.